rmw_connext_shared_cpp/test/test_security_logging.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 Canonical Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <string>
+
+#include "rmw/error_handling.h"
+#include "rmw_connext_shared_cpp/security_logging.hpp"
+
+#include "gmock/gmock.h"
+
+using ::testing::HasSubstr;
+
+namespace
+{
+// Environment variable names
+const char log_file_variable_name[] = "ROS_SECURITY_LOG_FILE";
+const char log_publish_variable_name[] = "ROS_SECURITY_LOG_PUBLISH";
+const char log_verbosity_variable_name[] = "ROS_SECURITY_LOG_VERBOSITY";
+
+// Connext-specific property names
+const char log_file_property_name[] = "com.rti.serv.secure.logging.log_file";
+const char verbosity_property_name[] = "com.rti.serv.secure.logging.log_level";
+const char distribute_enable_property_name[] = "com.rti.serv.secure.logging.distribute.enable";
+
+void custom_setenv(const std::string & variable_name, const std::string & value)
+{
+#ifdef _WIN32
+  auto ret = _putenv_s(variable_name.c_str(), value.c_str());
+#else
+  auto ret = setenv(variable_name.c_str(), value.c_str(), 1);
+#endif
+  if (ret != 0) {
+    ADD_FAILURE() << "Unable to set environment variable: expected 0, got " << ret;
+  }
+}
+
+const char * lookup_property_value(DDS::PropertyQosPolicy & policy, const char * property_name)
+{
+  auto property = DDS::PropertyQosPolicyHelper::lookup_property(
+    policy,
+    property_name);
+
+  if (property == nullptr) {
+    return nullptr;
+  }
+
+  return property->value;
+}
+
+const char * log_file_property(DDS::PropertyQosPolicy & policy)
+{
+  return lookup_property_value(policy, log_file_property_name);
+}
+
+const char * verbosity_property(DDS::PropertyQosPolicy & policy)
+{
+  return lookup_property_value(policy, verbosity_property_name);
+}
+
+const char * logging_distribute_enable_property(DDS::PropertyQosPolicy & policy)
+{
+  return lookup_property_value(policy, distribute_enable_property_name);
+}
+
+class SecurityLoggingTest : public ::testing::Test
+{
+public:
+  void SetUp()
+  {
+    custom_setenv(log_file_variable_name, "");
```

Comment:
There's a difference between the environment variable being set to `""` and not being set at all.

I suggest `SetUp()` ensures that the variables are *unset*. And then we can have additional tests for when variables are set to empty strings.

Reply:
Is there actually a difference? Unless I'm mistaken it wouldn't increase coverage at all here, but it _would_ require more platform-specific ifdefs.

> And then we can have additional tests for when variables are set to empty strings.

As far as I can tell, such adding more tests to test against empty strings or not set would in fact be tests against [`rcutils_get_env()`](https://github.com/ros2/rcutils/blob/master/include/rcutils/get_env.h#L32), which claims to return an empty string either way. Those types of tests should probably exist in rcutils, not here.

Reply:
ah, I see. Since rcutils is returning an empty string either way, disregard this comment.